### ISSUE TITLE
PGO error while building with clang: Could not read profile default.profdata: No such file or directory

### ISSUE THREAD

**Issue by recolic** [matrix-construct/construct#185]
Using clang 11.1.0 in ArchLinux. 

```
export CC=clang
export CXX=clang++
./configure --prefix=$HOME/tmp/something --with-included-rocksdb
make
```

And got the following error: 
```
Making all in include/ircd
make[1]: Entering directory '/home/recolic/tmp/construct/include/ircd'
(CDPATH="${ZSH_VERSION+.}:" && cd ../.. && /usr/bin/autoheader)
rm -f stamp-h1
touch config.h.in
cd ../.. && /bin/sh ./config.status include/ircd/config.h
config.status: creating include/ircd/config.h
config.status: include/ircd/config.h is unchanged
make  all-am
make[2]: Entering directory '/home/recolic/tmp/construct/include/ircd'
make[2]: Leaving directory '/home/recolic/tmp/construct/include/ircd'
make[1]: Leaving directory '/home/recolic/tmp/construct/include/ircd'
Making all in ircd
make[1]: Entering directory '/home/recolic/tmp/construct/ircd'
merge -output=default.profdata default.proftext
make[1]: merge: No such file or directory
make[1]: [Makefile:1349: default.profdata] Error 127 (ignored)
make  all-am
make[2]: Entering directory '/home/recolic/tmp/construct/ircd'
depbase=`echo assert.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../libtool  --tag=CXX   --mode=compile clang++ -std=gnu++17 -DHAVE_CONFIG_H -DIRCD_UNIT    -I../include -include ircd/ircd.pic.h   -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wwrite-strings -Wparentheses -Wundef -Wpacked -Wformat -Wformat-y2k -Wformat-nonliteral -Wstrict-aliasing=2 -Wstrict-overflow=5 -Wdisabled-optimization -Winvalid-pch -Winit-self -Wuninitialized -Wunreachable-code -Wno-overloaded-virtual -Wnon-virtual-dtor -Wctor-dtor-privacy -Wsign-promo -Wnull-dereference -Wundef -Wodr -Werror=return-type -Wno-missing-field-initializers -Wno-unused -Wno-unused-function -Wno-unused-label -Wno-unused-value -Wno-unused-variable -Wno-unused-parameter -Wno-endif-labels -Wno-unknown-pragmas -Wno-unknown-attributes -Wno-profile-instr-unprofiled -Wno-psabi -Wno-mismatched-tags -Wno-format-security -Wno-unknown-warning-option  -DPCH -DNDEBUG  -ftls-model=initial-exec -fno-c++-static-destructors -flto=thin -fstrict-vtable-pointers -fwhole-program-vtables -fprofile-use  -fsigned-char -mtune=native -march=native -mno-sse4a -fno-threadsafe-statics -fvisibility-inlines-hidden -fasynchronous-unwind-tables -grecord-gcc-switches -ggdb -g -fno-function-sections -fno-data-sections -O3 -pipe    -MT assert.lo -MD -MP -MF $depbase.Tpo -c -o assert.lo assert.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  clang++ -std=gnu++17 -DHAVE_CONFIG_H -DIRCD_UNIT -I../include -include ircd/ircd.pic.h -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wwrite-strings -Wparentheses -Wundef -Wpacked -Wformat -Wformat-y2k -Wformat-nonliteral -Wstrict-aliasing=2 -Wstrict-overflow=5 -Wdisabled-optimization -Winvalid-pch -Winit-self -Wuninitialized -Wunreachable-code -Wno-overloaded-virtual -Wnon-virtual-dtor -Wctor-dtor-privacy -Wsign-promo -Wnull-dereference -Wundef -Wodr -Werror=return-type -Wno-missing-field-initializers -Wno-unused -Wno-unused-function -Wno-unused-label -Wno-unused-value -Wno-unused-variable -Wno-unused-parameter -Wno-endif-labels -Wno-unknown-pragmas -Wno-unknown-attributes -Wno-profile-instr-unprofiled -Wno-psabi -Wno-mismatched-tags -Wno-format-security -Wno-unknown-warning-option -DPCH -DNDEBUG -ftls-model=initial-exec -fno-c++-static-destructors -flto=thin -fstrict-vtable-pointers -fwhole-program-vtables -fprofile-use -fsigned-char -mtune=native -march=native -mno-sse4a -fno-threadsafe-statics -fvisibility-inlines-hidden -fasynchronous-unwind-tables -grecord-gcc-switches -ggdb -g -fno-function-sections -fno-data-sections -O3 -pipe -MT assert.lo -MD -MP -MF .deps/assert.Tpo -c assert.cc  -fPIC -DPIC -o .libs/assert.o
error: Could not read profile default.profdata: No such file or directory
1 error generated.
make[2]: *** [Makefile:898: assert.lo] Error 1
make[2]: Leaving directory '/home/recolic/tmp/construct/ircd'
make[1]: *** [Makefile:692: all] Error 2
make[1]: Leaving directory '/home/recolic/tmp/construct/ircd'
make: *** [Makefile:540: all-recursive] Error 1
```

**Comment by jevolk** [comment]
I believe on many distributions including Arch clang uses a separate package for its runtime profiler. I'm not certain of this, but please check your packages in clang's domain.

As an aside, `./configure` should be catching this (though it is by blanking the environment variable, causing the command to become `merge` which is actually the first argument). Makefile could also catch this unset variable and skip the command possibly too. These both need to be fixed and if you make progress please submit a PR.

If not I will eventually address this, hopefully by the next tag release.

**Comment by jevolk** [comment]
You can now use `--with-profile=no` as a workaround. Leaving this open though until that's not required.

**Comment by jevolk** [comment]
Closed by f516b86e4d4aeeee828f795f0f05e08223cd4352
